kdc/fast.c

```diff
@@ -799,8 +799,10 @@ _kdc_fast_strengthen_reply_key(astgs_request_t r)
 
 	ret = krb5_generate_random_keyblock(r->context, r->reply_key.keytype,
 					    &r->strengthen_key);
-	if (ret)
-	    krb5_abortx(r->context, "random generator fail");
```

Comment:
I'm of the opinion that the KDC should definitely `abort()` if there's no RNG.  The RNG shouldn't fail though.

Reply:
`krb5_generate_random_block()` does `krb5_abortx()` if `krb5_generate_random()` fails, and `krb5_generate_random_keyblock()` calls `krb5_generate_random_block()`.  We should rationalize this a bit.

Reply:
`krb5_generate_random_block()` is `void`-returning.  We should deprecate it and not use it in-tree.